worknenjoy/gitpay
# Accept work feature to approve the start of the work #564

## Description

Now after been Assigned for a task the interested user need to confirm (reject, accept) the task.
User can accept, reject offer by e-mail (trigger action automatically by URL) or by accessing the task and clicking on the button.

## Changes

- New back end API end-point for assignment.
- New migration Adding new Column in Assign 

## Issue

> #564 

## Impacted Area

> Front-end Task
> Back-end API, routers, models

## Steps to test

Create a user
Import a new issue
Create another user
Apply to work on an issue
Log with the first user
Choose the user you created to start to work
The notification should be sent


## Assignment flow
### Receive notification that you are interested to work in the task
![reject-flow (1)](https://user-images.githubusercontent.com/35773631/80023167-cffc0a00-84e5-11ea-97a1-d36bfce12079.png)
### Task Owner notification
![reject-flow (2)](https://user-images.githubusercontent.com/35773631/80023170-d12d3700-84e5-11ea-8ce0-4c1a374d73ef.png)
### Task Owner Accept Offer
![reject-flow (3)](https://user-images.githubusercontent.com/35773631/80023171-d12d3700-84e5-11ea-97ce-3ef273d0f4aa.png)
## Update information (pending user )
![reject-flow (4)](https://user-images.githubusercontent.com/35773631/80023172-d1c5cd80-84e5-11ea-8204-9a42ccb0f41b.png)
### User option to accept/deny task on the task
![reject-flow (5)](https://user-images.githubusercontent.com/35773631/80023174-d1c5cd80-84e5-11ea-8b96-742e2454a47d.png)
### User option to accept/deny task on the email (links trigger option automatically)
![reject-flow (6)](https://user-images.githubusercontent.com/35773631/80023175-d25e6400-84e5-11ea-860f-ba7a5880f623.png)
## If deny pop up a modal asking for a motive
![reject-flow (7)](https://user-images.githubusercontent.com/35773631/80023177-d25e6400-84e5-11ea-997f-edb9780dfa11.png)
## After reject task updated properly, only owner and user have information
![reject-flow (8)](https://user-images.githubusercontent.com/35773631/80023179-d2f6fa80-84e5-11ea-928d-702e3fdbe5f1.png)
## Notification send to the Task Onwer if user reject Assign
![reject-flow (9)](https://user-images.githubusercontent.com/35773631/80023181-d2f6fa80-84e5-11ea-9db6-fd1454feb13e.png)
## User can re-send offer if rejected
![reject-flow (10)](https://user-images.githubusercontent.com/35773631/80023184-d38f9100-84e5-11ea-9f3a-598fc18822ec.png)
## If user accept Assign 
![Screenshot (13)](https://user-images.githubusercontent.com/35773631/80023702-9bd51900-84e6-11ea-9913-f7af04f5ced8.png)
## Onwer notification if User accept work in the task
![Screenshot (14)](https://user-images.githubusercontent.com/35773631/80023704-9c6daf80-84e6-11ea-83bc-461f8345dfb8.png)
